contracts/SwapAndBridge.sol

```diff
@@ -0,0 +1,150 @@
+        // would be passed in as constructor args. They are left here to help debugging and testing.
+        spokePool = USSSpokePoolInterface(0x6f26Bf09B1C792e3228e5467807a900A503c0281);
+        oneInchRouter = I1InchAggregationRouterV5(0x11111112542D85B3EF69AE05771c2dCCff4fAa26);
+    }
+
+    /**
+     * @notice Swaps tokens on this chain via 1Inch and bridges them via Across atomically. Caller can fully specify
+     * their slippage tolerance for the swap and also the full Across deposit params.
+     * @param aggregationExecutor Address of 1inch contract that executes calls described in `oneInchData`.
+     * @param swapDescription 1Inch SwapDescription struct, packed to consolidate function params:
+     *     - srcToken: Token to pull from msg.sender into contract and swap on 1inch into dstToken.
+     *     - dstToken: Token to receive from 1inch swap and to be deposited into Across.
+     *     - srcReceiver: Address to receive `srcToken` from msg.sender. Overwritten by this function to be this contract.
+     *     - dstReceiver: Address to receive `dstToken` from 1inch swap. Overwritten by this function to be this contract.
+     *     - amount: Amount of `srcToken` to swap. Return amount is deposited into Across.
+     *     - minReturnAmount: Minimum amount of `dstToken` to receive from 1inch swap.
+     *     - flags: Overwritten by this contract.
+     * @param outputToken Token to receive from Across deposit on destination chain.
+     * @param outputAmount of `outputToken` to receive from Across deposit on
+     * destination chain.
+     */
+    function swap1InchAndBridge(
+        IAggregationExecutor aggregationExecutor,
+        I1InchAggregationRouterV5.SwapDescription memory swapDescription,
+        address outputToken,
+        uint256 outputAmount,
+        address depositor,
+        address recipient,
+        uint256 destinationChainid,
+        address exclusiveRelayer,
+        uint32 quoteTimestamp,
+        uint32 fillDeadline,
+        uint32 exclusivityDeadline,
+        bytes memory message
+    ) external nonReentrant {
+        // @dev: Don't let caller use WETH, or better, hardcode the swap and acrossInput tokens to be USDC.e/USDC
+        // because 1Inch has special rules to handle WETH<-->ETH.
+
+        // Pull tokens from caller into this contract.
+        address swapToken = address(swapDescription.srcToken);
+        uint256 swapTokenAmount = swapDescription.amount;
+        IERC20(swapToken).transferFrom(msg.sender, address(this), swapTokenAmount);
+
+        // Craft 1Inch swap arguments to swap `swapToken` for `acrossInputToken` using this contract as the recipient
+        // of the swap.
+        swapDescription.srcReceiver = payable(address(this));
+        swapDescription.dstReceiver = payable(address(this));
+        swapDescription.flags = 0; // TODO: Change this if we want to support ETH swaps and/or partial fills (i.e.
```

Comment:
So 1inch guarantees if we set this to 0 that partial fills won't happen (for instance, if one of the trade paths reverts, for instance)?

Reply:
I'm not sure if they guarantee it but that's how I interpret their contract code